third_party/ti_simplelink_sdk/cc13x2_26x2/crypto/aes_alt.c

```diff
@@ -0,0 +1,176 @@
+/******************************************************************************
+
+ @file aes_alt.c
+
+ @brief AES implientation for TI chip
+
+ Group: CMCU, LPC
+ Target Device: cc13x2_26x2
+
+ ******************************************************************************
+ 
+ Copyright (c) 2017-2020, Texas Instruments Incorporated
```

Comment:
Generally third_party only contains only external repositories.

Could we add a repository link here instead of adding files as-is? Otherwise I would expect this to also have a CHIP Authors license (which I am guessing would not apply here).

Reply:
We don't maintain an external repository of our code, but some form of these files exists in our Software Development Kit. Most of the sources in this directory are direct copies from that SDK, but with development kit agnostic names. Historically all the board support files were named after the development kits the examples targeted, not the SoC families.

I am pushing to have those sources available in a generic location within the SDK, but that will require a new version to be released.

The next hurdle would be our porting layer. To have RTOS agnostic code TI made a POSIX-like porting layer for TI-RTOS and FreeRTOS. I had a bit of difficulty building this porting layer with GN because I could not remove the `--specs=nosys.spec` option from the compile.

If it were possible to build the POSIX porting layer [here](https://github.com/project-chip/connectedhomeip/blob/f200ae3fd7d57e542a0f733aab821d2e379fdafc/third_party/ti_simplelink_sdk/ti_simplelink_sdk.gni#L176)
most of the source files could be pulled from the TI SDK directly. Some of them would have to be pulled from a board specific example folder.

Reply:
I've updated the code to build the platform files from one of the example applications within the TI SDK. This is a bit more coupled to the examples directory in the SDK than I would prefer, but it works. This also requires one manual change to the platform files when the SDK is installed due to a difference in OpenThread version.